perception/autoware_compare_map_segmentation/src/voxel_based_compare_map_filter/node.cpp

```diff
@@ -62,6 +71,60 @@ VoxelBasedCompareMapFilterComponent::VoxelBasedCompareMapFilterComponent(
   RCLCPP_INFO(this->get_logger(), "tf_map_input_frame: %s", tf_input_frame_.c_str());
 }
 
+// TODO(badai-nguyen): Temporary Implementation: Delete this override function when autoware_utils
+// refactor (https://github.com/autowarefoundation/autoware_utils/pull/50) or new
+// ManagedTransformBuffer lib is deployed for autoware
+void VoxelBasedCompareMapFilterComponent::input_indices_callback(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr cloud, const PointIndicesConstPtr indices)
+{
+  // If cloud is given, check if it's valid
+  if (!isValid(cloud)) {
+    RCLCPP_ERROR(this->get_logger(), "[input_indices_callback] Invalid input!");
+    return;
+  }
+  // If indices are given, check if they are valid
+  if (indices && !isValid(indices)) {
+    RCLCPP_ERROR(this->get_logger(), "[input_indices_callback] Invalid indices!");
+    return;
+  }
+
+  // Check whether the user has given a different input TF frame
+  tf_input_orig_frame_ = cloud->header.frame_id;
+  PointCloud2ConstPtr cloud_tf;
+  if (cloud->header.frame_id != tf_input_frame_) {
+    RCLCPP_DEBUG(
+      this->get_logger(), "[input_indices_callback] Transforming input dataset from %s to %s.",
+      cloud->header.frame_id.c_str(), tf_input_frame_.c_str());
+    // Save the original frame ID
+    // Convert the cloud into the different frame
+    PointCloud2 cloud_transformed;
+
+    try {
+      // Lookup the transform from input frame to "map"
+      geometry_msgs::msg::TransformStamped transform_stamped = tf_buffer_.lookupTransform(
+        tf_input_frame_, tf_input_orig_frame_, rclcpp::Time(cloud->header.stamp),
+        rclcpp::Duration::from_seconds(1.0));
+
+      // Transform the point cloud
+      tf2::doTransform(*cloud, cloud_transformed, transform_stamped);
+      cloud_transformed.header.frame_id = tf_input_frame_;  // Update frame ID to "map"
+      cloud_tf = std::make_shared<PointCloud2>(cloud_transformed);
+    } catch (tf2::TransformException & ex) {
+      RCLCPP_WARN(this->get_logger(), "Could not transform point cloud: %s", ex.what());
+      cloud_tf = cloud;  // Fallback to original data
```

Comment:
This callback runs at `10 Hz`, so the `_THROTTLE` family of log functions should be used (e.g. log once only every 5s).

Reply:
@mojomex fixed here https://github.com/autowarefoundation/autoware_universe/pull/10299/commits/106514e67c51bf1578280579fab22b582e0f34dd